robolectric/src/main/java/org/robolectric/util/ActivityController.java

```diff
@@ -241,4 +241,26 @@ public void run() {
         .resume()
         .visible();
   }
+  
+  public ActivityController<T> configurationChange(final Application application, final Configuration newConfiguration) {
+    final T activity = get();
+    final Configuration currentConfig = activity.getResources().getConfiguration();
+    final int changedBits = currentConfig.diff(newConfiguration);
+    currentConfig.setTo(newConfiguration);
+    
+    // Can the activity handle itself ALL configuration changes?
+    if ((getActivityInfo(application).configChanges & changedBits) == changedBits) {
+      shadowMainLooper.runPaused(new Runnable() {
+        @Override
+        public void run() {
+          ReflectionHelpers.callInstanceMethod(Activity.class, component, "onConfigurationChanged",
+            ClassParameter.from(Configuration.class, newConfiguration));
+        }
+      });
+    } else {
+      Shadows.shadowOf(activity).recreateWithConfigChanges(changedBits);
```

Comment:
The two fields are both package private on Activity and you can set them directly via reflection with

ReflectionHelpers.setField(Activity.class, component, "mChangingConfigurations", true);
ReflectionHelpers.setField(Activity.class, component, "mConfigChangeFlags", changedBits);

Obviously its not great to use reflection, but this way is consistent with the code above and avoids adding a new public method to the Shadow.

Also in this case, I think we want to actually have a new instance of the Activity as mentioned here:-

https://developer.android.com/reference/android/app/Activity.html#onRetainNonConfigurationInstance()